pkg/3scale/amp/operator/system_mysql_options_provider.go

```diff
@@ -202,3 +203,11 @@ func (s *SystemMysqlOptionsProvider) podTemplateLabels() map[string]string {
 
 	return labels
 }
+
+func (s *SystemMysqlOptionsProvider) setPriorityClassNames() {
+	if s.apimanager.Spec.System.DatabaseSpec != nil &&
+		s.apimanager.Spec.System.DatabaseSpec.MySQL != nil &&
```

Comment:
`s.apimanager.Spec.System.DatabaseSpec.MySQL` is not necessarily not nil to have Mysql deployed, as Mysql is the default. 

It is better to call `apìmanager.IsSystemMysqlEnabled()`

Reply:
priorityClassname is not set for the system-mysql component

Reply:
update done to use IsSystemMysqlEnabled.
Regarding "priorityClassname is not set for the system-mysql component" - it's implemented, but I will retest it. Thank you for comments

Reply:
my bad, sorry. When `SystemMySQLReconciler` is called, it can safely be assumed mysql is enabled. 

I think it was right as it was

```go
if s.apimanager.Spec.System.DatabaseSpec != nil &&
		s.apimanager.Spec.System.DatabaseSpec.MySQL != nil &&
                s.apimanager.Spec.System.DatabaseSpec.MySQL.PriorityClassName != nil {
                s.mysqlOptions.PriorityClassName = *s.apimanager.Spec.System.DatabaseSpec.MySQL.PriorityClassName
}
```